### PR TITLE
Add description method to NSObject

### PR DESCRIPTION
**Pull request by visigoth**
This adds the `description` method to `NSObject`, which invokes the
`description` selector.

### CODE REVIEW

**Comment by visigoth** [comment]
this adds a dependency between `objc` and `foundation` due to `NSString`, which if you really want to keep separate, needs to be compiled conditionally on whether the `foundation` feature is enabled. should this go in the `foundation` module instead?

File: src/objc/ns_object.rs [hunk 1]
```diff
@@ -86,4 +87,10 @@ impl<'data> NSObject<'data> {
     pub fn mutable_copy(&self) -> Arc<Self> {
         unsafe { _msg_send_any_cached![self, mutableCopy] }
     }
+
+    /// Returns a string that describes the contents of this object.
+    #[inline]
+    pub fn description(&self) -> Arc<NSString> {
```
Comment (nvzqz) [inline src/objc/ns_object.rs]: This should be behind the `foundation` feature flag, to not require `foundation` when only using `objc`.

I also try to link to Apple's original documentation when I can.

```suggestion
    ///
    /// See [documentation](https://developer.apple.com/documentation/objectivec/1418956-nsobject/1418746-description?language=objc).
    #[cfg(feature = "foundation")]
    #[inline]
    pub fn description(&self) -> Arc<crate::foundation::NSString> {
```

**Review by nvzqz — CHANGES_REQUESTED** [review]
Thanks for the contribution! I'm surprised I didn't have this already. 🙂

See my comment for what needs changing.

While you're at it, I think it would be good to also do `debugDescription`.

File: src/objc/ns_object.rs (hunk 1 above)
Comment (visigoth) [reply src/objc/ns_object.rs]: alright, i made those changes. i mistakenly clobbered the original commit, but added new ones for the suggested changes. do you want me to build a commit history from the original PR commit, keep what's in the PR now, or should i squash before your merge?

**Comment by nvzqz** [comment]
Thanks for the changes! Sorry it took me so long to get back to this. Expect a release with this within a week 🙂

**Pull request merged**